src/Algebra/Morphism/Bundles.agda

```diff
@@ -0,0 +1,176 @@
+  magmaHomomorphism = record { isMagmaHomomorphism = isMagmaHomomorphism }
+
+  open MagmaHomomorphism magmaHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between Groups
+------------------------------------------------------------------------
+
+record GroupHomomorphism (A : Group a ℓa) (B : Group b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Group A
+    module B = Group B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isGroupHomomorphism : IsGroupHomomorphism A.rawGroup B.rawGroup ⟦_⟧
+
+  open IsGroupHomomorphism isGroupHomomorphism public
+
+  monoidHomomorphism : MonoidHomomorphism A.monoid B.monoid
+  monoidHomomorphism = record { isMonoidHomomorphism = isMonoidHomomorphism }
+
+  open MonoidHomomorphism monoidHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between NearSemirings
+------------------------------------------------------------------------
+
+record NearSemiringHomomorphism (A : NearSemiring a ℓa) (B : NearSemiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = NearSemiring A
+    module B = NearSemiring B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isNearSemiringHomomorphism : IsNearSemiringHomomorphism A.rawNearSemiring B.rawNearSemiring ⟦_⟧
+
+  open IsNearSemiringHomomorphism isNearSemiringHomomorphism public
+
+  +-monoidHomomorphism : MonoidHomomorphism A.+-monoid B.+-monoid
+  +-monoidHomomorphism = record { isMonoidHomomorphism = +-isMonoidHomomorphism }
+
+  open MonoidHomomorphism +-monoidHomomorphism public
+
+  *-magmaHomomorphism : MagmaHomomorphism A.*-magma B.*-magma
+  *-magmaHomomorphism = record { isMagmaHomomorphism = *-isMagmaHomomorphism }
+
+------------------------------------------------------------------------
+-- Morphisms between Semirings
+------------------------------------------------------------------------
+
+record SemiringHomomorphism (A : Semiring a ℓa) (B : Semiring b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = Semiring A
+    module B = Semiring B
+
+  field
+    ⟦_⟧ : A.Carrier → B.Carrier
+
+    isSemiringHomomorphism : IsSemiringHomomorphism A.rawSemiring B.rawSemiring ⟦_⟧
+
+  open IsSemiringHomomorphism isSemiringHomomorphism public
+
+  *-monoidHomomorphism : MonoidHomomorphism A.*-monoid B.*-monoid
+  *-monoidHomomorphism = record { isMonoidHomomorphism = *-isMonoidHomomorphism }
+
+  open MonoidHomomorphism *-monoidHomomorphism public
+
+------------------------------------------------------------------------
+-- Morphisms between RingWithoutOnes
+------------------------------------------------------------------------
+
+record RingWithoutOneHomomorphism (A : RingWithoutOne a ℓa) (B : RingWithoutOne b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
+  private
+    module A = RingWithoutOne A
+    module B = RingWithoutOne B
+
+  field
+    ⟦_⟧ : _ → _ -- A.Carrier → B.Carrier causes an disambiguation error!?
```

Comment:
```agda
Cannot resolve overloaded projection Carrier because
it is not applied to a visible argument
  candidates in scope:
- Carrier : Set a
- Carrier : Set a
when checking that the expression
Algebra.Morphism.Bundles.A.Carrier has type _226
```
hoping this is simply a stupid error on my part, but I can't see what...